### DISCUSSION TITLE
are there any limitations for various S3 storage classes (standard, glacier, etc)?

### DISCUSSION THREAD

**Discussion by onetimecontributor** [io7m-com/exfilac#32] [Q&A] [answered]
are there any compatibility limitations of the tool with various S3 storage classes (standard, glacier, one zone, etc)? 
I've checked docs, but as I'm not that knowledgeable in the area, I couldn't spot anything that looks like limitation for the compatible type of s3 storage class.
pls advise

**Comment by io7m** [comment, marked as answer]
Right now, the app will always specify the `STANDARD` class. This is for no other reason than having one fewer configuration knobs. It wouldn't be difficult to make it support all of the other classes.

**Reply by onetimecontributor** [reply]
tnx for your answer. have you by chance calculated previously is there a reasonable cost saving for using those pure "backup" options like glacier deep archive? 
I haven't yet figured out how s3 calc works to reflect my particular case, but I guess it will be similar to yours re photo storage.

**Reply by onetimecontributor** [reply]
also, from what you know about s3, is it just simple change of some parameter to switch to other classes (glacier), so one can fork your solution without deep rework? no guarantees ofc

**Reply by io7m** [reply]
I haven't calculated the cost differences as I don't actually use AWS myself (although I do test on it for compatibility).

I either use a local `minio` instance, or I use S3-compatible storage from other providers who don't distinguish between the storage classes.

If you need to fork, these are the requests you'd need to change (the builders have parameters for storage classes):

https://github.com/io7m-com/exfilac/blob/dc60a51ba7628d400bceb1d256e1da78a5cd2ec2/com.io7m.exfilac.s3_uploader.amazon/src/main/java/com/io7m/exfilac/s3_uploader/amazon/EFS3AMZUpload.kt#L211

https://github.com/io7m-com/exfilac/blob/dc60a51ba7628d400bceb1d256e1da78a5cd2ec2/com.io7m.exfilac.s3_uploader.amazon/src/main/java/com/io7m/exfilac/s3_uploader/amazon/EFS3AMZUpload.kt#L300

**Reply by io7m** [reply]
(To be clear, you get `STANDARD` if you don't specify anything, which is what the code currently does).
